c/busybox-1.22.0/cal_false-unreach-call_true-no-overflow_true-valid-memsafety.c

```diff
@@ -1832,9 +1832,41 @@ ssize_t write(int fildes, const void *buf, size_t nbyte)
   return ret;
 }
 
+int getopt_long(int argc, char * const argv[], const char *optstring,
+                const struct option *longopts, int *longindex)
+{
+  (void)*longopts;
+  (void)longindex;
+
+  int result = -1;
+
+  if(optind >= argc || argv[optind][0] != '-')
+    return -1;
+
+  size_t opt_index = __VERIFIER_nondet_ulong();
+  __VERIFIER_assume(opt_index < strlen(optstring) && optstring[opt_index] != ':');
+
+  if(__VERIFIER_nondet_int())
+  {
+    result = optstring[opt_index];
+    if(__VERIFIER_nondet_int())
+      ++optind;
+  }
+
+  if(result != -1 && optind < argc && optstring[opt_index+1] == ':')
+  {
+    optarg = argv[optind];
```

Comment:
Just a thought:
What if some optstring has an optional argument (possibly) missing? Then we should set optarg to NULL, right? There are definitely some checks for optarg being NULL in the benchmarks (but I haven't studied the code deeper, so maybe this is not a problem), e.g. https://github.com/sosy-lab/sv-benchmarks/blob/master/c/busybox-1.22.0/uniq_true-no-overflow_true-valid-memsafety.c#L819